### DISCUSSION TITLE
Assistance Needed: Connecting SAP ECC for Data Extraction Using pyrfc

### DISCUSSION THREAD

**Discussion by sandipsable506** [SAP/PyRFC#355] [Q&A] [answered]
Hello Everyone,

I am currently working on connecting SAP ECC for data extraction using pyrfc but encountering the following error:

```python
ImportError: cannot import name 'Connection' from partially initialized module 'pyrfc' (most likely due to a circular import)
```

I have not yet installed the SAP NWRFC SDK. Could this error be attributed to the absence of the SDK?

I am using SAP 770 GUI, and I am uncertain about which SAP NWRFC SDK version to install. I checked the SAP Store but couldn't find an SDK specifically for SAP 770. I have added the 64-bit RFC control while installing SAP 770 (refer to the snapshot ).

Are there any specific ABAP settings required on the SAP GUI?

Any guidance or links for the correct SAP NWRFC SDK version would be greatly appreciated.

Thanks in advance for your help.

Best regards,

Sandip
![64bitControlsSetup](https://github.com/SAP/PyRFC/assets/146274777/92b113ba-54a4-40ca-bdad-64d898f4f315)

**Comment by bsrdjan** [comment, marked as answer]
Hello Sandip,

SAP NW RFC SDK for PyRFC is not necessarily the same version used in SAP Logon Client. PyRFC releases and SAP Logon Client releases are not related, thus PyRFC can't "rely" on SAP NW RFC SDK files shipped with SAP Logon Client.

SAP NW RFC SDK is backwards compatible and PyRFC requires the latest version, see 
https://github.com/SAP/PyRFC?tab=readme-ov-file#requirements

The SAP NW RFC SDK installation is described here: https://sap.github.io/PyRFC/install.html

Installation as described should fix the issue 

Regards, Srdjan
